src/Xenial.Tasty.Cli/Commands/StudioCommand.cs

```diff
@@ -0,0 +1,188 @@
+                                });
+
+                                var sw = Stopwatch.StartNew();
+                                var exitCode = await commander.BuildProject(path, progress, cancellationTokenSource.Token);
+                                ((IProgress<(string line, bool isRunning, int exitCode)>)progress).Report(($"Finished in {sw.Elapsed}", true, exitCode));
+                            }
+                        }
+                    }
+                    ),
+                    new MenuItem ("_Quit", "", () => Application.RequestStop() )
+                }),
+                new MenuBarItem("_Color Scheme", CreateColorSchemeMenuItems()),
+                new MenuBarItem("_About...", "About Tasty.Cli", () => MessageBox.Query("About Tasty.Cli", aboutMessage.ToString(), "_Ok")),
+            });
+
+
+
+            _statusBar = new StatusBar(new[]
+            {
+                new StatusItem(Key.ControlQ, "~CTRL-Q~ Quit", () =>
+                {
+                    Application.RequestStop();
+                }),
+                new StatusItem(Key.ControlD, "~CTRL-D~ Debug", () =>
+                {
+                    if (!Debugger.IsAttached)
+                    {
+                        Debugger.Launch();
+                    }
+                }),
+                new StatusItem(Key.ControlC, "~CTRL-C~ Cancel", () =>
+                {
+                    cancellationTokenSource.Cancel();
+                }),
+                new StatusItem(Key.F9, "~F9~ Open Menu", () =>
+                {
+                    _menu.OpenMenu();
+                }),
+            });
+
+            _top.Add(_menu, _leftPane, _rightPane, _statusBar);
+
+            SetColorScheme();
+
+            Application.Run();
+            return Task.FromResult(1);
+        }
+
+        static void SetColorScheme()
+        {
+#pragma warning disable CS8601 // Possible null reference assignment.
+            _leftPane!.ColorScheme = _baseColorScheme;
+#pragma warning restore CS8601 // Possible null reference assignment.
+            // _rightPane.ColorScheme = _baseColorScheme;
+            _top?.SetNeedsDisplay();
+        }
+
+        static ColorScheme? _baseColorScheme;
+        static MenuItem[] CreateColorSchemeMenuItems()
+        {
+            List<MenuItem> menuItems = new List<MenuItem>();
+            foreach (var sc in Colors.ColorSchemes)
+            {
+                var item = new MenuItem();
+                item.Title = sc.Key;
+                item.CheckType |= MenuItemCheckStyle.Radio;
+
+#pragma warning disable CS8604 // Possible null reference argument.
+                item.Checked = sc.Value == _baseColorScheme;
+#pragma warning restore CS8604 // Possible null reference argument.
+
+                item.Action += () =>
+                {
+                    _baseColorScheme = sc.Value;
+                    SetColorScheme();
+                    foreach (var menuItem in menuItems)
```

Comment:
Is this an implicit capture of menuItems?

Reply:
It is. I'm totally not happy with the UI code in general, is copied directly from the terminal gui showcase demo. Not sure about this is a problem though

Reply:
Under the hood, menuItems get lifted as a field to a generated class which holds the delegate as a method.
This now prevents the list of menuItems from being garbage collected, you are creating a circular reference without a root here.

Reply:
Cause the cli is not supposed to be reused yet, i think it's fine, cause it's at the very start of the app. The application will close at a point anyway now.